### DISCUSSION TITLE
SOFAHelper compilation error

### DISCUSSION THREAD

**Discussion by zbounik** [sofa-framework/sofa#2775] [Build / Config / Environment] [answered]
Hi,

In building SOFA 21.12 on Ubuntu 18.04, I am receiving this error: 

/home/zahra/sofa/v21.12/src/SofaKernel/modules/SofaHelper/src/sofa/helper/system/FileSystem.cpp:26:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
SofaKernel/SofaFramework/SofaHelper/CMakeFiles/Sofa.Helper.dir/build.make:536: recipe for target 'SofaKernel/SofaFramework/SofaHelper/CMakeFiles/Sofa.Helper.dir/src/sofa/helper/system/FileSystem.cpp.o' failed
make[2]: *** [SofaKernel/SofaFramework/SofaHelper/CMakeFiles/Sofa.Helper.dir/src/sofa/helper/system/FileSystem.cpp.o] Error 1
CMakeFiles/Makefile2:2312: recipe for target 'SofaKernel/SofaFramework/SofaHelper/CMakeFiles/Sofa.Helper.dir/all' failed
make[1]: *** [SofaKernel/SofaFramework/SofaHelper/CMakeFiles/Sofa.Helper.dir/all] Error 2
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2

I appreciate any idea on how to fix this problem.

Zahra

**Comment by fredroy** [comment, marked as answer]
Hello @zbounik 
Ubuntu 20.04 is needed for SOFA v21.12.
Your error comes from the fact that the STL header `filesystem` was not supported officially with the gcc shipped with Ubuntu 18.04.

So you can either:
 - upgrade your ubuntu version
 - change the compiler on your own. I dont remember but I think the clang shipped with Ubuntu18.04 was OK (not sure though)

**Reply by zbounik** [reply]
AAAA ok then! Thank you. 
I tried the clang it doesn't work.

**Reply by LianhaoYin** [reply]
Tried with Ubuntu 20.04, it dose not work

**Reply by fredroy** [reply]
What version of SOFA are you trying to compile ?
FYI, This discussion was about SOFA 21.12 (+2 years ago)

**Reply by LianhaoYin** [reply]
my problem is solved by changing gcc and g++ from 7 to 9.  Sofa version: 23.12, ubuntu 20.04
